www/src/py_utils.js

```diff
@@ -1347,6 +1347,27 @@ $B.rich_comp = function(op, x, y){
         }
     }
     var res, rev_op, compared = false
+
+    if(x.__class__ && x.__class__.$factory === $B.$type.$factory && y.__class__ && y.__class__.$factory === $B.$type.$factory) {
+        if ( op == '__eq__') {
+            return (x === y || x.$factory === y || y.$factory === x)
```

Comment:
I am not exactly sure why `x === y` is not enough, but without this the descriptor tests fail. What is the correct way to test if two types are equal?